### DISCUSSION TITLE
Add new Command "ShutterSetPosition"

### DISCUSSION THREAD

**Discussion by jwiesel** [arendst/Tasmota#10345] [Ideas]
Hi there,
I installed my first Shelly 2.5 to control my shutters. Great feature of Tasmota! :-)
Unfortunately I am missing something in my specific setup

# My Use Case
* I have a central device controlling all shutters and one device per shutter to open/close it. (part of the "pre-smart" electrical installation in my home"
* Technically this is implemented by connecting two inputs for each shutter. One to controlled all shutters from the central instance + one to control a specific shutter via the switch right next to the window.
* I installed a shelly 2.5 running tasmota right next to the window
* The central switch is also smart and is used to open / close all shutters during the morning/evening.

# Issue
* As two different devices control the actual position of the shutter (e.g. 80% open), I need to synchronize the shutter position in Tasmota (update via external input without initiating a shutter move by Tasmota)

# Solution Approach
* I added a flow in NodeRed to check, if the central device moves the shutters.
* After it stopped moving, I reset the ShutterPosition in Tasmota depending on the actual shutter position.
* Unfortunately I only found the two commands "ShutterSetOpen" and "ShutterSetClose" to either set the position to fully open or fully closed

# Feature Request
* Can you provide a new command "ShutterSetPosition" to reset the shutter to an exact position (e.g. 80%) instead of 0 or 100%?
* By that I could synchronize my two devices controlling the shutter

Thanks for your help!

**Comment by jcisio** [comment]
There is already the `ShutterPosition<x>` command as described here https://tasmota.github.io/docs/Commands/#shutters. But I think it is bugged. In my test:

```
14:23:39 CMD: ShutterPosition1 50
14:23:39 MQT: stat/volet1/RESULT = {"POWER2":"ON"}
14:23:39 MQT: stat/volet1/POWER2 = ON
14:23:39 MQT: stat/volet1/RESULT = {"ShutterPosition1":50}
14:23:39 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":99,"Direction":-1,"Target":50}}
14:23:40 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":97,"Direction":-1,"Target":50}}
14:23:41 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":92,"Direction":-1,"Target":50}}
14:23:42 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":86,"Direction":-1,"Target":50}}
14:23:43 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":81,"Direction":-1,"Target":50}}
14:23:44 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":76,"Direction":-1,"Target":50}}
14:23:45 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":71,"Direction":-1,"Target":50}}
14:23:46 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":65,"Direction":-1,"Target":50}}
14:23:47 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":60,"Direction":-1,"Target":50}}
14:23:48 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":55,"Direction":-1,"Target":50}}
14:23:48 MQT: stat/volet1/RESULT = {"POWER2":"ON"}
14:23:48 MQT: stat/volet1/POWER2 = ON
14:23:49 MQT: stat/volet1/SHUTTER1 = 50
14:23:49 MQT: stat/volet1/RESULT = {"Shutter1":{"Position":50,"Direction":0,"Target":50}}
```
I don't know what is Position -1 (it should be 0 or 1), and the shutter does not stop at that position as expected. Idem, when I use the slider in the web UI to change position, it does not stop at the desired position.

**Comment by jwiesel** [comment]
Hi @jcisio ,
thanks for your reply! I actually want to change the shutter-position only in Tasmota without actually moving the physical shutter.
The situation occurs, if the shutter position in tasmota (let's say 80% open) does not match reality / the physical shutter (let's say 40%)

**Reply by ortegafernando** [reply]
I think it will be a good command. I had similar case. I was doing some test with tasmota, and the motor of my blind stops (due to the heat) just inside a movement. After some minutes to cold down, I could move again my shutter. 

So my tasmota had a different position than the actual/physical position. I need to open the junction box and move the blind without tasmota, directly with "cables".

It will be great to have a command to say tasmota: "hey guy!!! , my actual/current/physical position is XX%"

I think it makes sense.

**Reply by jcisio** [reply]
Manual synchronization won't be precise. I think in your case (send command to open at 80%, blocked at 40%), the easiest way is to send 0% (it will be 0% in both Tasmota and in reality) then send 80%.

**Reply by ortegafernando** [reply]
> Manual synchronization won't be precise. I think in your case (send command to open at 80%, blocked at 40%), the easiest way is to send 0% (it will be 0% in both Tasmota and in reality) then send 80%.

Imagine that you have your shutter/blind in a real 50%, but Tasmota says 0% (for example because the motor has stop). 
That way YOU CAN'T close to 0% your shutter/blind.
But if you could say to Tasmota that your real position in that moment is 50% or 60% or whatever, after that you could ask Tasmota to move to 0% and you will be able to close completely your shutter/blind.
After, you will have to calibrate it again, but at least, you can close your shutter/blind without opening any electrical box.
Do you understand me ?

Thanks.

**Reply by jcisio** [reply]
I think I understand. For this example, if it is 50% real and 0% Tasmota, then just open to 100%, it will be 100% real and 100% Tasmota. I never needed to open the box.

**Reply by ortegafernando** [reply]
> I think I understand. For this example, if it is 50% real and 0% Tasmota, then just open to 100%, it will be 100% real and 100% Tasmota. I never needed to open the box.

Great, thanks.

**Reply by stefanbode** [reply]
One of the drivers biggest benefit is, that it know where the shutter is and what current configuration (e.g. tilt or last direction) is. It was never made to operate from the outside by operating the motor without giving tasmota an idea that there is a change. Additionally: If you can operate in % your shutter from the outside, why do you use the shutter mode. In this case just operate on two relays and done. Even, if adding this command is not a big deal it would implicate that this is a supported operation which it is not. setopen and setclose are only for callibration and should never used during normal operation. Additionally the internal accuracy about different shutter parameters is higher than the 0..100%. This would additionally cause side-effects. If you can control your shutter from an automation software, please just configure two relays in TASMOTA and do not use the shutter driver
